ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/sorter/AbstractPendingTransactionsSorter.java

```diff
@@ -138,6 +141,39 @@ public void evictOldTransactions() {
             });
   }
 
+  /**
+   * Evaluates the transaction pool and evicts future nonce transactions until the global count of
+   * future nonce transactions is below the configured limit.
+   *
+   * <p>Eviction criteria is first distance from the current account nonce, and then age in the
+   * pool.
+   *
+   * @param worldState worldstate for account lookup
+   */
+  public void evictGlobalFutureTransactions(final MutableWorldState worldState) {
+    final var senderAccountMap = new HashMap<Address, Account>();
+
+    pendingTransactions.values().stream()
+        // filter pending transactions for future nonce transactions
+        .filter(
+            txInfo ->
+                senderAccountMap.computeIfAbsent(txInfo.getSender(), worldState::get).getNonce()
```

Comment:
Neat use of map here.

nit: I think it's possible to only store the nonce instead of the whole account in the map without impacting readability.